src/challenges.js

```diff
@@ -22,6 +31,24 @@ const cacheFunction = cb => {
   // If the returned function is invoked with arguments that it has already seen
   // then it should return the cached result and not invoke `cb` again.
   // `cb` should only ever be invoked once for a given set of arguments.
+  const cache = {};
+  let i = 0;
+  let duplicate = false;
+
+  return (...args) => {
+    duplicate = Object.keys(cache).forEach(key => {
+      if (cache[key] === args) return true;
+    });
+
```

Comment:
I love this approach you've taken here. I love being about to walk through a solution and see a step by step clean approach.